File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/digitaltwin/ObjectServiceContext.java
package org.opcfoundation.webserver.types.digitaltwin;

import org.opcfoundation.webserver.types.common.UaObjectId;

import java.util.ArrayList;
import java.util.List;

public class ObjectServiceContext {
    private final UaObjectId objectId;

    public ObjectServiceContext(UaObjectId objectId)
    {
        this.objectId = objectId;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public String getCurrentObjectName()
    {
        return (null == objectId.getInstance()) ? objectId.getId() : objectId.getInstance().browseName();
    }

    /*
    public List<String> getCurrentObjectPath()
    {
        // To be implemented
        return new ArrayList<>();
    }*/
}

File: opcua-webservice-node/src/server/types/digitaltwin/ObjectServiceContext.ts
import { UaObjectId } from "../common/UaObjectId";

export class ObjectServiceContext {
    private readonly _objectId: UaObjectId;

    constructor(objectId: UaObjectId) {
        this._objectId = objectId;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    getCurrentObjectName(): string {
        return this._objectId.instance?.browseName ?? this._objectId.id;
    }
}